examples/glm_reporter_example/full_examples.py

```diff
@@ -0,0 +1,195 @@
+# contrasts['EV2>EV1'] = contrasts['EV2'] - contrasts['EV1']
+# contrasts['effects_of_interest'] = contrasts['EV1'] + contrasts['EV2']
+#
+# """fit GLM"""
+# print('\r\nFitting a GLM (this takes time) ..')
+# fmri_glm = FirstLevelGLM()
+# fmri_glm.fit(fmri_files, design_matrix)
+#
+# """save computed mask"""
+# mask_path = os.path.join(subject_data.output_dir, "mask.nii.gz")
+# print("Saving mask image %s" % mask_path)
+# nibabel.save(fmri_glm.masker_.mask_img_, mask_path)
+#
+# # compute bg unto which activation will be projected
+# mean_fmri_files = compute_mean_3D_image(fmri_files)
+# print("Computing contrasts ..")
+# z_maps = {}
+# for contrast_id, contrast_val in contrasts.items():
+#     print("\tcontrast id: %s" % contrast_id)
+#     z_map, t_map, eff_map, var_map = fmri_glm.transform(
+#             con_vals=contrasts[contrast_id],
+#             contrast_name=contrast_id,
+#             output_z=True,
+#             output_stat=True,
+#             output_effects=True,
+#             output_variance=True)
+#
+#     # store stat maps to disk
+#     for dtype, out_map in zip(['z', 't', 'effects', 'variance'],
+#                               [z_map, t_map, eff_map, var_map]):
+#         map_dir = os.path.join(
+#                 subject_data.output_dir, '%s_maps' % dtype)
+#         if not os.path.exists(map_dir):
+#             os.makedirs(map_dir)
+#         map_path = os.path.join(
+#                 map_dir, '%s.nii.gz' % contrast_id)
+#         nibabel.save(out_map, map_path)
+#
+#         if dtype == "z":
+#             z_maps[contrast_id] = map_path
+#
+#         print("\t\t%s map: %s" % (dtype, map_path))
+#
+#     print
+#
+#     from pprint import pprint
+#
+#     pprint(contrasts)
+#
+#     anat_file
+#
+#     """do stats report"""
+#     reports_dir = os.path.join(subject_data.output_dir, "reports")
+#     stats_report_filename = os.path.join(reports_dir, "report_stats.html")
+#     contrasts = dict((contrast_id, contrasts[contrast_id])
+#                      for contrast_id in z_maps.keys())
+#     generate_subject_stats_report(
+#             stats_report_filename,
+#             contrasts,
+#             z_maps,
+#             fmri_glm.masker_.mask_img_,
+#             design_matrices=[design_matrix],
+#             subject_id=subject_data.subject_id,
+#             anat=anat_file,
+#             cluster_th=50,  # we're only interested in this 'large' clusters
+#             start_time=stats_start_time,
+#
+#             # additional ``kwargs`` for more informative report
+#             paradigm=paradigm,
+#             TR=TR,
+#             n_scans=n_scans,
+#             hfcut=hfcut,
+#             frametimes=frametimes,
+#             drift_model=drift_model,
+#             hrf_model=hrf_model,
+#             slicer='z'
+#             )
+#
+#     # shutdown main report page
+#     ProgressReport().finish_dir(output_dir)
+#
+#     print("\r\nStatistic report written to %s\r\n" % stats_report_filename)
```

Comment:
???

Reply:
This was causing problems and doesn't really matter right now for my purposes so I commented it out for now and will look at this later, once more pieces are in place.